x/oracle/client/cli/query.go

```diff
@@ -18,5 +22,149 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
 		RunE:                       client.ValidateCmd,
 	}
 
+	cmd.AddCommand(
+		GetCmdQueryAggregatePrevote(),
+		GetCmdQueryAggregateVote(),
+		GetCmdQueryParams(),
+	)
+
+	return cmd
+}
+
+// GetCmdQueryParams implements the query params command.
+func GetCmdQueryParams() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "params",
+		Args:  cobra.NoArgs,
+		Short: "Query the current Oracle params",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			res, err := queryClient.Params(context.Background(), &types.QueryParamsRequest{})
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+// GetCmdQueryAggregateVote implements the query aggregate prevote of the validator command
+func GetCmdQueryAggregateVote() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "aggregate-votes [validator]",
+		Args:  cobra.RangeArgs(0, 1),
+		Short: "Query outstanding oracle aggregate votes.",
+		Long: strings.TrimSpace(`
+Query outstanding oracle aggregate vote.
+
+$ umeed query oracle aggregate-votes
+
+Or, can filter with voter address
+
+$ umeed query oracle aggregate-votes umeevaloper...
+`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			if len(args) == 0 {
+				res, err := queryClient.AggregateVotes(
+					context.Background(),
+					&types.QueryAggregateVotesRequest{},
+				)
+				if err != nil {
+					return err
+				}
+
+				return clientCtx.PrintProto(res)
+			}
+
+			valString := args[0]
+			validator, err := sdk.ValAddressFromBech32(valString)
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.AggregateVote(
+				context.Background(),
+				&types.QueryAggregateVoteRequest{ValidatorAddr: validator.String()},
+			)
+			if err != nil {
+				return err
+			}
```

Comment:
I would just keep this and remove the above. Instead, you can just construct the query object based on user input. e.g.

```go
query := &types.QueryAggregateVoteRequest{}
if len(args) > 0 {
  // ...
  query.ValidatorAddr = ...
}
```

This way, you remove some duplicate code.

Also, typically args are not really optional and if they are, we should consider making it a flag instead, e.g. `--validator=...`